src/ar/hentaislayer/src/eu/kanade/tachiyomi/extension/ar/hentaislayer/HentaiSlayer.kt

```diff
@@ -0,0 +1,245 @@
+package eu.kanade.tachiyomi.extension.ar.hentaislayer
+
+import android.app.Application
+import android.widget.Toast
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.util.Calendar
+
+class HentaiSlayer : ParsedHttpSource(), ConfigurableSource {
+
+    override val name = "هنتاي سلاير"
+
+    override val baseUrl = "https://hentaislayer.net"
+
+    override val lang = "ar"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Referer", "$baseUrl/")
+        .set("Origin", baseUrl)
+
+    private val preferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    // ============================== Popular ===============================
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/manga?page=$page", headers)
+
+    override fun popularMangaSelector() = "div > div:has(div#card-real)"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        element.selectFirst("div#card-real a")?.run {
+            setUrlWithoutDomain(absUrl("href"))
+            selectFirst("figure")?.run {
+                selectFirst("img.object-cover")?.run {
+                    thumbnail_url = imgAttr()
+                    title = attr("alt")
+                }
+                genre = select("span p.drop-shadow-sm").text()
+            }
+        }
+        genre = element.select("span a[href*='?genre=']")
+            .map { it.text() }
+            .let { listOf(genre) + it }
```

Comment:
consider saving the genre from above in a variable and building the genre string once.

or just remove genre from here since you are parsing them in `mangadetails` as well